File: python/src/projects/skale_manager/skale_manager_project.py
from instance import Instance
from network import Network
from project import Project, ProjectMetadata
from skale_manager_instance import SkaleManagerInstance


class SkaleManagerProject(Project):
    def __init__(self, network: Network, metadata: ProjectMetadata) -> None:
        super().__init__(network, metadata)
        self.githubRepo = 'https://github.com/skalenetwork/skale-manager/'

    def create_instance(self, address: str) -> Instance:
        return SkaleManagerInstance(self, address)

    def get_abi_filename(version: str):
        return f'skale-manager-{version}-abi.json'

File: typescript/src/projects/skale-manager/skaleManagerProject.ts
import { Instance } from "../../instance";
import { Project } from "../../project";
import { SkaleManagerInstance } from "./skaleManagerInstance";

export class SkaleManagerProject<ContractType, InterfaceType> extends
    Project<ContractType, InterfaceType> {
    githubRepo = "https://github.com/skalenetwork/skale-manager/";

    createInstance (address: string): Instance<ContractType, InterfaceType> {
        return new SkaleManagerInstance(
            this,
            address
        );
    }

    getAbiFilename (version: string) {
        return `${this.metadata.name}-${version}-abi.json`;
    }
}
